src/compiler/jsgen.js

```diff
@@ -559,11 +567,19 @@ class JSGenerator {
         case 'op.less': {
             const left = this.descendInput(node.left);
             const right = this.descendInput(node.right);
+            // When the left operand is a number or NaN and the right operand is a number, we can use <
+            if (left.isAlwaysNumberOrNaN() && right.isAlwaysNumber()) {
+                return new TypedInput(`(${left.asNumberOrNaN()} < ${right.asNumber()})`, TYPE_BOOLEAN);
+            }
+            // When the left operand is a number and the right operand is a number or NaN, we can negate >=
+            if (left.isAlwaysNumber() && right.isAlwaysNumberOrNaN()) {
+                return new TypedInput(`!(${left.asNumber()} >= ${right.asNumberOrNaN()})`, TYPE_BOOLEAN);
+            }
             // When both operands are known to never be numbers, only use string comparison to avoid all number parsing.
             if (left.isNeverNumber() || right.isNeverNumber()) {
```

Comment:
Shouldn't you use `&&`?

Reply:
No, the logic is correct and that line isn't even changed by this PR.

Reply:
If either are never numbers then I believe it should be safe to convert them to strings and compare them, as non-numbers can't really do weird things like `"0.000123" == "1.23e-4"`

Reply:
Maybe try `parseInt()` and assume values are not a number if an error is thrown?